pkg/sink/codec/common/decoder.go

```diff
@@ -40,3 +49,146 @@ type RowEventDecoder interface {
 	// NextDDLEvent returns the next DDL event if exists
 	NextDDLEvent() (*commonEvent.DDLEvent, error)
 }
+
+func appendCol2Chunk(idx int, raw interface{}, ft types.FieldType, chk *chunk.Chunk) {
+	mysqlType := ft.GetType()
```

Comment:
this cannot be shared among protocols, since each protocol encode data in different format.

we can simply implement the similar logic for all protocols, and then combine them together.